backend/src/main/kotlin/com/github/quillraven/toilapp_dataloader/DataLoaderApplication.kt

```diff
@@ -0,0 +1,144 @@
+        "obsessed",
+        "pannikin"
+)
+
+@Configuration
+@EnableAutoConfiguration
+@EnableReactiveMongoRepositories(basePackages = ["com.github.quillraven.toilapp.repository"])
+@ComponentScan(
+        basePackages = ["com.github.quillraven.toilapp.model",  "com.github.quillraven.toilapp.repository",  "com.github.quillraven.toilapp.service",
+            "com.github.quillraven.toilapp_dataloader"]
+)
+class DataLoaderApplication
+
+@Component
+class DataLoaderRunner(
+        @Autowired private val imgService: ImageService,
+        @Autowired private val toiletService: ToiletService,
+        @Autowired private val userService: UserService,
+        @Autowired private val context: ConfigurableApplicationContext
+): CommandLineRunner {
+
+    val numToilets = 20
+
+    override fun run(vararg args: String?) {
+        createUsers().flatMap { createToilets(it) }.subscribe{
+            println("$numToilets toilets created..")
+            exitProcess(SpringApplication.exit(context))
+        }
+    }
+
+    private fun createUsers(): Mono<List<Any>> {
+        val userMonoList = mutableListOf<Mono<User>>()
+        //create users
+        for(usrName in userNames) {
+            val userName = userNames[Random.nextInt(0, userNames.size)]
+            val email = "$userName@mail.com"
+            val user = User(UUID.randomUUID().toString(), userName, email)
+            userMonoList.add(userService.createUser(user))
+        }
+        return Mono.zip(userMonoList){ uArr -> uArr.toList() }
+    }
+
+    private fun createToilets(uList: List<Any>): Mono<List<Any>> {
+        val toiletMonoList = mutableListOf<Mono<Toilet>>()
+        for(num in 0 until numToilets) {
+            toiletMonoList.add(createToilet(uList))
+        }
+        return Mono.zip(toiletMonoList){ tArr -> tArr.toList()}
+    }
+
+    private fun createToilet(uList: List<Any>): Mono<Toilet> {
+        val imgName = "toilet" +  Random.nextInt(1, 10) + ".jpg"
+        val title = titles[Random.nextInt(0, titles.size)]
+        val description = descriptions[Random.nextInt(0, descriptions.size)]
+        val rating = Random.nextDouble(1.0, 6.0)
+        val numCommands = Random.nextInt(0, 4)
+        val comments = mutableListOf<Comment>()
+        val log = Random.nextDouble(10.0, 20.0)
+        val lat = Random.nextDouble(45.0, 55.0)
+
+        val userList = mutableListOf<User>()
+        userList.addAll(uList as List<User>)
+        for(i in 0..numCommands) {
+            val user = userList.get(Random.nextInt(0, userList.size))
+            val comment = Comment(UUID.randomUUID().toString(), user, Date(), commentTexts[Random.nextInt(0, commentTexts.size)])
+            comments.add(comment)
+        }
+
+        val inStreamCallable: Callable<InputStream> = Callable<InputStream>{ DataLoaderRunner::class.java.getResourceAsStream("/sample-images/$imgName")}
+        val objId = imgService.store(inStreamCallable, "my-name")
+        return objId.flatMap{oid -> run{
```

Comment:
`run` is not needed here. I remember you mentioned an error that you got when removing it? What error do you get? Because for me it is working if I remove it.